Remove warnings about autocreated index fields

scollazo opened the pull request:
Removes models.W042 warning when running manage.py commands

Connected to https://github.com/archivematica/Issues/issues/1759

scollazo committed:
Set DEFAULT_AUTO_FIELD

Removes models.W042 warning when running manage.py commands

replaceafill commented on src/archivematica/storage_service/storage_service/settings/base.py:
```diff
@@ -746,3 +746,6 @@ def _get_email(email):
     LOGIN_EXEMPT_URLS.append(r"^metrics$")
+
+# Remove models.W042 warning
+DEFAULT_AUTO_FIELD = "django.db.models.AutoField"
```
See https://github.com/artefactual/archivematica/pull/2097#discussion_r2288326605

scollazo committed:
Use SILENCED_SYSTEM_CHECKS in production settings

This removes the warnings about autocreated primary keys

scollazo commented:
@replaceafill , I removed the defalt_auto_field and used systemd_silence_checks only for production. Does this look better?

replaceafill commented:
@scollazo I think instead of hardcoding the setting value we should expose the `SILENCED_SYSTEM_CHECKS` Django setting the same way expose other Django settings: with [application variables](https://github.com/artefactual/archivematica/blob/8df5aeeadcdd23b58946b36df45b11de75b60267/src/archivematica/dashboard/install/README.md#application-variables) in the Dashboard and [application-specific environment variables](https://github.com/artefactual/archivematica-storage-service/tree/122f1fcc823b119b089798599a4dec50e556ffca/install#application-specific-environment-variables) in the Storage Service. That way we could document their default values and explain where those codes (i.e. `models.W042`) come from and allow users to customize them.

scollazo committed:
Add SS_SILENCED_SYSTEM_CHECKS environment var

Based on https://docs.djangoproject.com/en/4.2/ref/settings/#silenced-system-checks

scollazo commented:
Done!

replaceafill commented on install/README.md:
```diff
@@ -223,6 +223,12 @@ of these settings or provide values to mandatory fields.
   - **Type:** `integer`
   - **Default:** `900`
 
+- **`SS_SILENCED_SYSTEM_CHECKS`**:
```
What do you think of dropping `SS_` prefix? We are not really consistent but most of the Django related variables don't have it. Also could we move this documentation bit after the `SECRET_KEY` variable?

replaceafill commented on install/README.md:
```diff
@@ -223,6 +223,12 @@ of these settings or provide values to mandatory fields.
   - **Default:** `900`
 
+- **`SS_SILENCED_SYSTEM_CHECKS`**:
+  - **Description:**  comma-separated list of ignored system checks.
+    e.g. mysql.W002,models.W042
```
Let's add a link to https://docs.djangoproject.com/en/4.2/ref/settings/#silenced-system-checks here. See examples in `TIME_ZONE` or `SECRET_KEY` above.

replaceafill commented on src/archivematica/storage_service/storage_service/settings/base.py:
```diff
@@ -754,3 +754,5 @@ def _get_email(email):
     INSTALLED_APPS = INSTALLED_APPS + ["django_prometheus"]
     LOGIN_EXEMPT_URLS.append(r"^metrics$")
+
+SILENCED_SYSTEM_CHECKS = environ.get("SS_SILENCED_SYSTEM_CHECKS", "[]")
```
If you're expecting the user to set a comma separated list it needs to be converted to a list here:

```suggestion
SILENCED_SYSTEM_CHECKS = environ.get("SS_SILENCED_SYSTEM_CHECKS", "").split(",")
```

replaceafill commented on install/README.md:
```diff
@@ -223,6 +223,12 @@ of these settings or provide values to mandatory fields.
+  - **Description:**  comma-separated list of ignored system checks.
+    e.g. mysql.W002,models.W042
+  - **Type:** `string`
+  - **Default:** `None`
```
```suggestion
  - **Default:** `""`
```
